### ISSUE TITLE
NextJS Microfrontends / Sharing Distribution

### ISSUE THREAD

**Issue by alexnaish** [anomalyco/sst#5071]
Firstly, thank you for all the work you guys put in - such a huge amount of work and it's very much appreciated.

I'm working on a project that'll have three distinct areas. I want to split out the codebase and deploy 3 distinct apps (ideally deployable independently) which all sit under the main domain. 

I've seen that this has been discussed and implemented earlier (https://github.com/sst/sst/pull/2348) but I can't see any reference to sharing distributions for Nextjs sites.

As it stands, if I wanted to deploy my three sites I would have to have three unique public Cloudfront distributions:

* Main Entry: Marketing / Public
* App 2: Private Dashboard
* App 3: Docs

I could deploy each currently and add rewrites in the main app to proxy /docs to the respective cloudfront or I could manually tweak the main distribution using the transform function (as in: https://sst.dev/docs/examples#aws-nextjs-add-behavior) to add each of the origins + behaviours for App 2 and 3 but I feel like that's delving too deep into the internals of how opennext / SST deploy a nextjs site and I'd be left with 2 publicly accessible cloudfront distributions which I don't need or want. 

Either way it feels off that I'd either have Cloudfront to Cloudfront routing or need to have an understanding of the inner workings of opennext deployments. If the SST NextJS construct allowed specifying a distribution ID we could have all the origins in one and tweak the origin behaviours based off the nextjs basePath.

What's your thoughts? Am I missing an easier alternative?

**Comment by jayair** [comment]
Yeah so the Nextjs component (and the other SSR sites) are meant to be standalone. It's a gap in how they are designed because their CF distributions need to be configured quite a bit.

Ideally we'd want to be able to use the router component in front of all of them and have their CloudFronts disabled. It's on the roadmap but it's quite a bit of work.

The other alternative is to run these as containers. I don't have an example for that handy.
